internal/app/agent/sender/sender.go

```diff
@@ -20,40 +28,82 @@ func NewSender(serverAddress string) *Sender {
 	}
 }
 
+func compressData(data []byte) (*bytes.Buffer, error) {
+	var buf bytes.Buffer
+	gz := gzip.NewWriter(&buf)
```

Comment:
[CHANGE REQUEST] It should be a middleware.

Reply:
Done